streams/src/main/java/org/apache/kafka/streams/kstream/internals/foreignkeyjoin/CombinedKey.java

```diff
@@ -39,9 +39,6 @@ public KP getPrimaryKey() {
     }
 
     public boolean equals(final KF foreignKey, final KP primaryKey) {
-        if (this.primaryKey == null) {
```

Comment:
This is removed because `primaryKey` is guaranteed in constructor to never be null.